tests/test_dir_layouts.py

```diff
@@ -0,0 +1,120 @@
+import pytest
+from pathlib import Path
+import shutil
+import bids
+from petdeface.petdeface import PetDeface
+from os import cpu_count
+from bids.layout import BIDSLayout
+
+import tempfile
+
+# collect test bids dataset from data directory
+data_dir = Path(__file__).parent.parent / "data"
+
+# get number of cores, use all but one
+nthreads = cpu_count() - 1
+
+layout = BIDSLayout(data_dir, validate=True)
+
+if layout:
+    pass
+
+
+def test_anat_in_first_session_folder():
+    # create a temporary directory to copy the existing dataset into
+    with tempfile.TemporaryDirectory() as tmpdir:
+        shutil.copytree(data_dir, Path(tmpdir) / "anat_in_first_session_folder")
+
+        # run petdeface on the copied dataset
+        petdeface = PetDeface(
+            Path(tmpdir) / "anat_in_first_session_folder",
+            output_dir=Path(tmpdir)
+            / "anat_in_first_session_folder_defaced"
+            / "derivatives"
+            / "petdeface",
+            n_procs=nthreads,
+        )
+        petdeface.run()
+
+
+def test_anat_in_each_session_folder():
+    # create a temporary directory to copy the existing dataset into
+    with tempfile.TemporaryDirectory() as tmpdir:
+        shutil.copytree(data_dir, Path(tmpdir) / "anat_in_each_session_folder")
+
+        # create a second session
+        second_session_folder = (
+            Path(tmpdir) / "anat_in_each_session_folder" / "sub-01" / "ses-second"
+        )
+        second_session_folder.mkdir(parents=True, exist_ok=True)
+
+        shutil.copytree(
+            Path(tmpdir) / "anat_in_each_session_folder" / "sub-01" / "ses-baseline",
+            second_session_folder,
+            dirs_exist_ok=True,
+        )
+
+        # replace the ses- entities in the files in the newly created second session folder
+        for file in second_session_folder.glob("pet/*"):
+            shutil.move(
+                file,
+                second_session_folder
+                / "pet"
+                / file.name.replace("ses-baseline_", "ses-second_"),
+            )
+
+        for file in second_session_folder.glob("anat/*"):
+            shutil.move(
+                file,
+                second_session_folder
+                / "anat"
+                / file.name.replace("ses-baseline_", "ses-second_"),
+            )
+
+        # run petdeface on the copied dataset
+        petdeface = PetDeface(
+            Path(tmpdir) / "anat_in_each_session_folder",
+            output_dir=Path(tmpdir)
+            / "anat_in_each_session_folder_defaced"
+            / "derivatives"
+            / "petdeface",
+            n_procs=nthreads,
+        )
+        petdeface.run()
+
```

Comment:
It would be good to check if the defaced PET file(s) is/are created.

Reply:
See my comment about the mask files, we're going to need to slightly modify the pipeline to output nifti masks instead of `mgz`, but that's irrelevant for the purposes of this branch/PR. I'd like to add all the tests even if it means everything is failing.